12-version_control.Rmd

```diff
@@ -0,0 +1,821 @@
+designed for project-specific communication: they both generally do not have facilities for organizing conversations by project subtopics,
+searching for conversations related to particular bugs or software versions, etc.
+
+GitHub *issues* are an alternative written communication medium to email and 
+messaging apps, and were designed specifically to facilitate project-specific 
+communication. Issues are *opened* from the "Issues" tab on the project's
+GitHub page, and they persist there even after the conversation is over and the issue is *closed* (in 
+contrast to email, issues are not usually deleted). One issue thread is usually created
+per topic, and they are easily searchable using GitHub's search tools. All 
+issues are accessible to all project collaborators, so no one is left out of 
+the conversation. Finally, issues can be setup so that team members get email 
+notifications when a new issue is created or a new post is made in an issue 
+thread. Replying to issues from email is also possible. Given all of these advantages,
+ we highly recommend the use of issues for project-related communication.
+
+To open a GitHub issue, 
+first click on the "Issues" tab (Figure \@ref(fig:issue-01)).
+
+```{r issue-01, fig.cap = 'The "Issues" tab on the GitHub web interface.', fig.retina = 2}
+image_read("img/version_control/issue_01.png") %>%
+  image_crop("3584x1250")
+```
+
+Next click the "New issue" button (Figure \@ref(fig:issue-02)).
+
+```{r issue-02, fig.cap = 'The "New issues" button on the GitHub web interface.', fig.retina = 2}
+image_read("img/version_control/issue_02.png") %>%
+  image_crop("3584x1250")
+```
+
+Add an issue title (which acts like an email subject line), and then put the 
+body of the message in the larger text box. Finally click "Submit new issue"
+to post the issue to share with others (Figure \@ref(fig:issue-03)).
+
+```{r issue-03, fig.cap = 'Dialog boxes and submission button for creating new GitHub issues.', fig.retina = 2}
+image_read("img/version_control/issue_03.png") %>%
+  image_crop("3584x2200")
+```
+
+You can reply to an issue that someone opened by adding your written response to
+the large text box and clicking comment (Figure \@ref(fig:issue-04)).
+
+```{r issue-04, fig.cap = 'Dialog box for replying to GitHub issues.', fig.retina = 2}
+image_read("img/version_control/issue_04.png") %>%
+  image_crop("3584x2000")
+```
+
+When a conversation is resolved, you can click "Close issue". 
+The closed issue can be later viewed be clicking the "Closed" header link 
+in the "Issue" tab (Figure \@ref(fig:issue-06)).
+
+```{r issue-06, fig.cap = 'The "Closed" issues tab on the GitHub web interface.', fig.retina = 2}
+image_read("img/version_control/issue_06.png") %>%
+  image_crop("3584x1900")
+```
+
+## Additional resources
+
+Now that you've picked up the basics of version control with Git and GitHub, you can expand your knowledge using 
```

Comment:
should we add specific things they should get out of each of these? I'm not too fussed about this but I think to be consistent with the other chapters...

Reply:
done a bit in 129a8de